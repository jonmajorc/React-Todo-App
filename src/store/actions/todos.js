import actionTypes from '../actionTypes'

export const addTodo = todo => ({
  type: actionTypes.ADD_TODO,
  todo,
})

export const toggleTodo = (id, isComplete) => ({
  type: actionTypes.TOGGLE_TODO,
  isComplete,
  id,
})

export const removeTodo = id => ({
  type: actionTypes.REMOVE_TODO,
  id,
})
