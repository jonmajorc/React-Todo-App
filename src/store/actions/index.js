import * as todoActions from './todos'

export default {
  ...todoActions,
}