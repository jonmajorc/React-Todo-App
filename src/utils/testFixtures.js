// modules
import React from 'react'
import { mount } from 'enzyme'
// import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { configureStoreTest } from '../store/configureStore'
import initialState from '../store/initialState'
import { createStore } from 'redux'
import rootReducer from '../store/reducers'
import { MemoryRouter as Router } from 'react-router-dom'
// components
import App from '../layouts/App'

/***************************************************************************\
  Object Mothers
\***************************************************************************/
export function renderAppComponent(props = {}, opts) {
  const { route = '', Comp = App } = opts

  const initialStoreState = {
    ...initialState,
    ...props,
  }

  let store, container

  const mockStore = configureStoreTest()
  store = createStore(rootReducer, initialStoreState)
  return {
    store,
    mount: mount(
      <Provider store={store}>
        <Router initialEntries={[route]}>
          <Comp />
        </Router>
      </Provider>
    ),
  }
}
