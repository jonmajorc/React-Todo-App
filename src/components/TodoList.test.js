// modules
import React from 'react'

// utils
import { renderAppComponent } from '../utils/testFixtures'

// components
import TodoListConnected, { TodoList } from './TodoList'
import actions from '../store/actions'

describe('TodoList', () => {
  it('renders a list of todos', () => {
    const todos = [
      {
        id: 0,
        item: 'clean car',
        isComplete: false,
      },
    ]

    const container = renderAppComponent(
      {
        todos,
      },
      { Comp: TodoListConnected }
    )
    const TodoListComponent = container.mount.find(TodoList)
    expect(TodoListComponent.prop('todos')).toEqual(todos)
  })

  it('renders a list of many todos with different keys', () => {
    const todos = [
      {
        id: 0,
        item: 'clean car',
        isComplete: false,
      },
      {
        id: 1,
        item: 'clean car',
        isComplete: false,
      },
    ]

    const container = renderAppComponent(
      {
        todos,
      },
      { Comp: TodoListConnected }
    )
    const TodoListComponent = container.mount.find(TodoList)
    expect(TodoListComponent.prop('todos')).toEqual(todos)
  })

  it('toggles completetions todos on click', () => {
    const selTodo = `[data-test="todo-item-1"]`
    const todos = [
      {
        id: 0,
        item: 'clean car',
        isComplete: false,
      },
      {
        id: 1,
        item: 'clean car',
        isComplete: false,
      },
    ]

    const container = renderAppComponent(
      {
        todos,
      },
      { Comp: TodoListConnected }
    )
    const TodoListComponent = container.mount.find(TodoList)

    let todo = TodoListComponent.find(selTodo)
    expect(todo).toMatchSnapshot()
    todo.simulate('click')
    expect(todo.instance()).toMatchSnapshot()
  })
})
