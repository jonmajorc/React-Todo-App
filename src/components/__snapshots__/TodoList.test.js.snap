// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList toggles completetions todos on click 1`] = `
<li
  className="todo-list__item"
  data-test="todo-item-1"
  key="1"
  onClick={[Function]}
>
  <button
    className="todo-list__item__btn--del"
    onClick={[Function]}
  >
    X
  </button>
  <span
    className="todo-list__item__text"
    data-completed={false}
  />
</li>
`;

exports[`TodoList toggles completetions todos on click 2`] = `
<li
  data-test="todo-item-1"
  class="todo-list__item"
>
  <button
    class="todo-list__item__btn--del"
  >
    X
  </button>
  <span
    class="todo-list__item__text"
    data-completed="true"
  />
</li>
`;
